trieste/models/keras/models.py

```diff
@@ -409,39 +409,59 @@ def log(self, dataset: Optional[Dataset] = None) -> None:
         summary_writer = logging.get_tensorboard_writer()
         if summary_writer:
             with summary_writer.as_default(step=logging.get_step_number()):
-                logging.scalar("num_epochs", len(self.model.history.epoch))
+                logging.scalar("epochs/num_epochs", len(self.model.history.epoch))
                 for k, v in self.model.history.history.items():
-                    logging.histogram(f"{k}_epoch", v)
-                    logging.scalar(f"{k}_final", v[-1])
-                    logging.scalar(f"{k}_diff", v[0] - v[-1])
-                    logging.scalar(f"{k}_min", tf.reduce_min(v))
-                    logging.scalar(f"{k}_max", tf.reduce_max(v))
+                    k_sep = k.split("_")
+                    if "loss" in k and "model" in k:
```

Comment:
This is very arcane! Without knowing the contents of history I can't tell what the names will be (and whether this will ever throw an exception due to k_sep being size 1). Maybe a few comments with examples?

Reply:
I was unhappy with this bit, feels a bit brittle to rely on output names assigned by KerasEnsemble - here I was trying to separate loss as a group and identify if any other metrics is in the output (here we have `mse` by default in `DeepEnsemble`) and if so make it a grouping

the alternative is to go with the unprocessed variant but then loss and whatever other metrics are defined will be mixed in ... perhaps I can get metrics names somehow from the compiled model, that would make it cleaner

Reply:
the default output looks like: 
```
{"loss":[...], "model_i_output_loss":[...], "model_i_output_mse":[...]}
```
where i is the member of ensemble, so we would have e.g. 10 of those

Reply:
Maybe use a regex based whitelist instead then? Clearer and more resilient.

```python
KEY_SPLITTER = {
    # map history keys to prefix and suffix
    "loss" : ("loss", ""),
    r"(?P<model>model_\d+)_output_loss" : ("loss", r"_\g<model>"),
    r"(?P<model>model_\d+)_(?P<metric>.+)" : (r"\g<metric>", r"_\g<model>"),
}
for pattern, (pre_sub, post_sub) in KEY_SPLITTER.items():
    if re.match(pattern, k):
        pre = re.sub(pattern, pre_sub, k)
        post = re.sub(pattern, post_sub, k)
        break
else:
    # unrecognised history key; ignore
    continue
```